also a partial nixpkgs-review gives
```
error: --- Error --- nix-daemon
builder for '/nix/store/6d9svxq7fdh008g25ly3gxhksqv6q1k3-luajit-2.1.0-2020-08-05-ltermbox-0.2-1.drv' failed with exit code 1; last 10 log lines:
    106 | static const struct luaL_reg lt_buffer_metatable[] = {
        |                              ^~~~~~~~~~~~~~~~~~~
  lua/ltermbox.c:195:30: error: array type has incomplete element type 'struct luaL_reg'
    195 | static const struct luaL_reg ltermbox[] = {
        |                              ^~~~~~~~
  lua/ltermbox.c:213:12: warning: return type defaults to 'int' [-Wimplicit-int]
    213 | LUALIB_API luaopen_ltermbox(lua_State *L) {
        |            ^~~~~~~~~~~~~~~~
  
  Error: Build error: Failed compiling object lua/ltermbox.o
note: keeping build directory '/tmp/nix-build-luajit-2.1.0-2020-08-05-luaexpat-1.3.0-1.drv-0'
error: --- Error --- nix-daemon
builder for '/nix/store/fbidsmb2sp9vsh3xlgq7q32p14warijr-luajit-2.1.0-2020-08-05-luaexpat-1.3.0-1.drv' failed with exit code 1; last 10 log lines:
  gcc -O2 -fPIC -I/nix/store/wbr4wh47z6vscz2d6a6yii9x9vnls8d4-luajit-2.1.0-2020-08-05/include/luajit-2.1 -c src/lxplib.c -o src/lxplib.o -I/nix/store/kqhd4kk5cgaj37xcswdmmfygpvww65ms-expat-2.2.8-dev/include -Isrc/
  src/lxplib.c:593:13: error: static declaration of 'luaL_setfuncs' follows non-static declaration
    593 | static void luaL_setfuncs (lua_State *L, const luaL_Reg *l, int nup) {
        |             ^~~~~~~~~~~~~
  In file included from src/lxplib.c:15:
  /nix/store/wbr4wh47z6vscz2d6a6yii9x9vnls8d4-luajit-2.1.0-2020-08-05/include/luajit-2.1/lauxlib.h:88:18: note: previous declaration of 'luaL_setfuncs' was here
     88 | LUALIB_API void (luaL_setfuncs) (lua_State *L, const luaL_Reg *l, int nup);
        |                  ^~~~~~~~~~~~~
  
  Error: Build error: Failed compiling object src/lxplib.o
note: keeping build directory '/tmp/nix-build-luajit-2.1.0-2020-08-05-luaffi-scm-1.drv-0'
error: --- Error --- nix-daemon
builder for '/nix/store/zsp5znkj605ib6dyr5r8qn0izbshx9pl-luajit-2.1.0-2020-08-05-luaffi-scm-1.drv' failed with exit code 1; last 10 log lines:
  ffi.h:88:13: error: static declaration of 'luaL_setfuncs' follows non-static declaration
     88 | static void luaL_setfuncs (lua_State *L, const luaL_Reg *l, int nup) {
        |             ^~~~~~~~~~~~~
  In file included from ffi.h:31,
                   from call.c:9:
  /nix/store/wbr4wh47z6vscz2d6a6yii9x9vnls8d4-luajit-2.1.0-2020-08-05/include/luajit-2.1/lauxlib.h:88:18: note: previous declaration of 'luaL_setfuncs' was here
     88 | LUALIB_API void (luaL_setfuncs) (lua_State *L, const luaL_Reg *l, int nup);
        |                  ^~~~~~~~~~~~~
  
  Error: Build error: Failed compiling object call.o
note: keeping build directory '/tmp/nix-build-luajit-2.1.0-2020-08-05-digestif-0.2-1.drv-0'
error: --- Error --- nix-daemon
builder for '/nix/store/44g3f848il1r8kks2vl70s2dfd8wkcs8-luajit-2.1.0-2020-08-05-digestif-0.2-1.drv' failed with exit code 1; last 10 log lines:
  patching sources
  configuring
  building
  installing
  Missing dependencies for digestif 0.2-1:
     lua >= 5.3 (5.1-1 provided by VM)
  
  digestif 0.2-1 depends on lua >= 5.3 (5.1-1 provided by VM)
```

--- a/pkgs/development/lua-modules/luaexpat.patch
+++ b/pkgs/development/lua-modules/luaexpat.patch
@@ -1,0 +1,36 @@
+diff --git a/src/lxplib.c b/src/lxplib.c
+index 1c972db..5712611 100644
+--- a/src/lxplib.c
++++ b/src/lxplib.c
+@@ -590,7 +590,7 @@ static void set_info (lua_State *L) {
+ /*
+ ** Adapted from Lua 5.2.0
+ */
+-static void luaL_setfuncs (lua_State *L, const luaL_Reg *l, int nup) {
++static void compat_luaL_setfuncs (lua_State *L, const luaL_Reg *l, int nup) {
+   luaL_checkstack(L, nup, "too many upvalues");
+   for (; l->name != NULL; l++) {  /* fill the table with given functions */
+     int i;
+@@ -602,6 +602,8 @@ static void luaL_setfuncs (lua_State *L, const luaL_Reg *l, int nup) {
+   }
+   lua_pop(L, nup);  /* remove upvalues */
+ }
++#else
++#define compat_luaL_setfuncs(L, reg, nup) luaL_setfuncs(L, reg, nup)
+ #endif
+ 
+ 
+@@ -612,11 +614,11 @@ int luaopen_lxp (lua_State *L) {
+ 	lua_pushvalue(L, -2);
+ 	lua_rawset(L, -3);
+ 
+-	luaL_setfuncs (L, lxp_meths, 0);
++	compat_luaL_setfuncs (L, lxp_meths, 0);
+ 	lua_pop (L, 1); /* remove metatable */
+ 
+ 	lua_newtable (L);
+-	luaL_setfuncs (L, lxp_funcs, 0);
++	compat_luaL_setfuncs (L, lxp_funcs, 0);
+ 	set_info (L);
+ 	return 1;
+ }

--- a/pkgs/development/lua-modules/overrides.nix
+++ b/pkgs/development/lua-modules/overrides.nix
@@ -204,6 +204,9 @@ with super;
     externalDeps = [
       { name = "EXPAT"; dep = pkgs.expat; }
     ];
+    patches = [
+      ./luaexpat.patch
+    ];
   });
 
   # TODO Somehow automatically amend buildInputs for things that need luaffi

